### PR TITLE
Feature/etcd-lock-manager

### PR DESCRIPTION
**Pull request by subhankarc**
- Adding ETCD LockManager
- Adding lock, unlock and isWriteLocked method
- Adding unit tests for the methods
- Adding constants file for eventmesh package
- Changing the config path
- Removing the eventmesh config
- adding new settings and certificates

### CODE REVIEW

File: test/eventmesh.LockManager.spec.js [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+          });
+      });
+
+      it('loc should not fail even if release of resource lock fails.', () => {
+        const noLockResp = {
+          'count': 0,
+          'operationType': ''
+        };
+        jsonStub.onCall(0).returns(noLockResp);
+        releaseStub.onCall(0).throws(new Error('Failed for acquire lock'));
```
Comment (ketakig12) [inline test/eventmesh.LockManager.spec.js]: Should it be "Failed to release lock"?

File: test/eventmesh.LockManager.spec.js [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+            /* jshint expr: true */
+            expect(lockStub.getCall(0).calledWithExactly('fakeResource/lock')).to.be.true;
+            expect(getStub.getCall(0).calledWithExactly('fakeResource/lock/details')).to.be.true;
+            expect(acquireStub.called).to.be.true;
+            expect(releaseStub.called).to.be.true;
+            sinon.assert.calledOnce(releaseStub);
+            expect(putStub.getCall(0).calledWithExactly('fakeResource/lock/details')).to.be.true;
+          });
+      });
+    });
```
Comment (ketakig12) [inline test/eventmesh.LockManager.spec.js]: Can you also add test case covering https://github.com/cloudfoundry-incubator/service-fabrik-broker/pull/274/files#diff-4f6777b6aa31c708f2f14b1e84688d2bR66

File: test/eventmesh.LockManager.spec.js [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+            expect(e.message).to.eql('Could not acquire lock for fakeResource as it is already locked.');
+            expect(lockStub.getCall(0).calledWithExactly('fakeResource/lock')).to.be.true;
+            expect(getStub.getCall(0).calledWithExactly('fakeResource/lock/details')).to.be.true;
+            expect(acquireStub.called).to.be.true;
+            expect(releaseStub.called).to.be.true;
+            sinon.assert.calledOnce(releaseStub);
+          });
+      });
+
+      it('loc should not fail even if release of resource lock fails.', () => {
```
[the remark this answers is not in the document]
Comment (subhankarc) [reply test/eventmesh.LockManager.spec.js]: corrected

File: test/eventmesh.LockManager.spec.js (hunk 1 above)
Comment (subhankarc) [reply test/eventmesh.LockManager.spec.js]: yes

File: test/eventmesh.LockManager.spec.js (hunk 2 above)
Comment (subhankarc) [reply test/eventmesh.LockManager.spec.js]: sure

**Comment by subhankarc** [comment]
@muralisc can you pls review this? We can merge it then.

**Pull request merged**